### PR TITLE
Chore: Update `@grafana/eslint-config`

### PR DESCRIPTION
**Pull request by leventebalogh**
### What changed?
Updated the `@grafana/eslint-config` to the latest version and also cleaned up the uneccessary ESLint related deps in our package.json.

### Why?
Why not? Also the outdated `eslint-plugin-jsdoc` in our deps required <17 versions of node to run the project with.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,19 +21,13 @@
...
     "@auto-it/first-time-contributor": "^10.37.6",
     "@auto-it/slack": "^10.37.6",
-    "@grafana/eslint-config": "^2.5.0",
+    "@grafana/eslint-config": "^5.1.0",
     "@grafana/tsconfig": "^1.2.0-rc1",
     "@swc/core": "^1.2.162",
     "@swc/jest": "^0.2.20",
     "@types/jest": "^27.4.1",
     "@types/node": "18.11.7",
-    "@typescript-eslint/eslint-plugin": "^4.33.0",
```
Comment (academo) [inline package.json:30]: Are you sure we don't need these dependencies anymore? I can see them as part of the plugins common package.json template.

Comment (jackw) [reply package.json:30]: I'm pretty sure they aren't needed as the dependencies in the package.json template are for after a plugin is scaffolded where as this worktree `package.json` has the common devDependencies for the workspaces `create-plugin` and `sign-plugin`. 

This fixes the node 18 install issue and cleans up as `@grafana/eslint-config` has most of these as nested deps too.

File: package.json [hunk 2]
```diff
@@ -21,19 +21,13 @@
...
-    "@typescript-eslint/parser": "^4.33.0",
     "auto": "^10.37.6",
-    "eslint": "^7.32.0",
-    "eslint-config-prettier": "^8.3.0",
-    "eslint-plugin-jsdoc": "^36.1.0",
-    "eslint-plugin-prettier": "^4.0.0",
```
Comment (jackw) [inline package.json:36]: Was this ever used? Asking as the latest eslint-config doesn't use it or include it.

**Review by jackw — APPROVED** [review]
Nice! 🚀 

Left a comment about the eslint-prettier-plugin.

File: package.json (hunk 2 above)
Comment (leventebalogh) [reply package.json:36]: That's a very good point, and I am not sure when it was used - maybe in the beginning. But it looks to me like we don't need it anymore 🤔

File: package.json (hunk 1 above)
Comment (leventebalogh) [reply package.json:30]: Yea, as Jack says those are the dependencies of the scaffolded plugins, while these are the dependencies of the tools themselves. I think we could clean up the deps of the scaffolded plugins as well, but that's a separate step.

**Pull request merged**